tiny_dnn/core/kernels/conv2d_op_opencl.h

```diff
@@ -20,14 +20,14 @@ class Conv2dOpenCLForwardOp : public core::OpKernel {
 #if defined(USE_OPENCL) || defined(USE_CUDA)
     auto params = OpKernel::params_->conv();
 
-    // incoming/outcoming data
-    const tensor_t &in_data = context.input(0);
-    const tensor_t &W       = context.input(1);
-    const tensor_t &bias    = context.input(2);
-    tensor_t &out_data      = context.output(1);
+    // TODO(Randl): Remove once layers forward and backward by themself.
+    Tensor<float_t> in_data_t(context.input(0));
+    const Tensor<float_t> weights_t(context.input(1)),
+      bias_t = Tensor<float_t>(context.input(2));  // TODO has_bias
```

Comment:
It's just my personal preferrence but declaring multiple variables in a single statement might spoil the readability in some cases.